frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -42,6 +50,56 @@ const updateGroupstate = (
   return newState;
 };
 
+// TODO: make an interface for the return value?
+const exclusiveProjectDependencies = (
+  state: State,
+  group: Group,
+  project: string
+): { upstreams: string[]; downstreams: string[] } => {
```

Comment:
debated whether I should define this in a `Dependency` interface but thought perhaps not needed for now? lmk if you prefer an interface tho